native/src/components/TtsContainer.tsx

```diff
@@ -72,42 +92,64 @@ const TtsContainer = ({ children }: TtsContainerProps): ReactElement => {
             KEY_PARAM_STREAM: 'STREAM_MUSIC',
           },
           iosVoiceId: '',
-          rate: 1,
+          rate: 0.45,
         })
+        setIsPlaying(true)
       }
     },
-    [languageCode, sentenceIndex, sentences],
+    [isPaused, languageCode, sentenceIndex, sentences, suppressFinishEvent],
   )
 
-  const stop = async () => {
+  const stop = useCallback(async () => {
+    await Tts.stop()
+    suppressFinishEventOnIos()
     setIsPlaying(false)
     setSentenceIndex(0)
-    await Tts.stop()
-    const TTS_STOP_DELAY = 100
-    await new Promise(resolve => {
-      setTimeout(resolve, TTS_STOP_DELAY)
-    })
-  }
+  }, [])
 
-  const pause = () => {
-    Tts.stop()
+  /** The stop function on ios triggers the wrong event ('tts-finish' instead of 'tts-cancel')
+   * 'tts-finish' triggers the next sentence, so we use pause which is only available on ios
+   */
+  const pause = async () => {
+    if (Platform.OS === 'ios') {
+      Tts.pause().then(() => setIsPaused(true))
+    } else {
+      Tts.stop()
+    }
     setIsPlaying(false)
   }
 
-  const playNext = useCallback(() => {
+  const playNextAutomatic = useCallback(() => {
     const nextIndex = sentenceIndex + 1
     if (nextIndex < sentences.length) {
       setSentenceIndex(nextIndex)
       play(nextIndex)
     } else {
-      stop()
+      stop().then()
+    }
+  }, [play, sentenceIndex, sentences.length, stop])
+
+  const playNext = useCallback(() => {
+    const nextIndex = sentenceIndex + 1
+    if (nextIndex < sentences.length) {
+      // do not update index  because ios immediately triggers playNextAuto due to `tts-finish` event, so it will be updated there
+      if (Platform.OS === 'ios') {
```

Comment:
also tried here use suppress function instead of not updating the index but it also leads to that the next sentence will be read and then it will be stopped